### PR TITLE
Remove Optional Arguments, Part 1

### PR DESCRIPTION
**Pull request by jasondegraw**
Pull request overview
---------------------
This is the first PR with more than a few optionals removed, there'll be several more on the way but this one has grown large enough that I'm concerned that it'll be hard to get it in without a lot of conflics. Most of these are replacing the optional argument with a regular argument with a default value. For example `Optional<std::string> arg = _` might now be `std::string_view arg = {}` or similar, it depends on the details of the case. This one grew large because I misread the usage count of `SetupOutputVariable` as 430 when it's more like 4300. Oops. In any case, there shouldn't be diffs, but I am expecting some formatting and conflict resolution.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Myoldmopar** [comment]
@jasondegraw this isn't conflicted right now, are you all set here?  If so, I can run Clang Format on this branch and get to reviewing it.

**Comment by jasondegraw** [comment]
@Myoldmopar That would be great if you could do that, much appreciated

File: src/EnergyPlus/InternalHeatGains.cc [hunk 1]
```diff
@@ -6516,16 +6516,16 @@ namespace InternalHeatGains {
...
                                 "Electricity",
                                 "InteriorEquipment",
                                 state.dataHeatBal->ZoneITEq(itEqNum).EndUseSubcategoryUPS,
                                 "Building",
                                 state.dataHeatBal->Zone(state.dataHeatBal->ZoneITEq(itEqNum).ZonePtr).Name,
                                 state.dataHeatBal->Zone(state.dataHeatBal->ZoneITEq(itEqNum).ZonePtr).Multiplier,
                                 state.dataHeatBal->Zone(state.dataHeatBal->ZoneITEq(itEqNum).ZonePtr).ListMultiplier,
-                                _,
-                                _,
+                                -999,
```
Comment (Myoldmopar) [inline src/EnergyPlus/InternalHeatGains.cc:6527]: Should this be DataSizing::AutoSize?  Or is this something else?

File: src/EnergyPlus/BaseboardElectric.cc [hunk 2]
```diff
@@ -355,10 +355,10 @@ namespace BaseboardElectric {
                                 OutputProcessor::SOVTimeStepType::System,
                                 OutputProcessor::SOVStoreType::Summed,
                                 thisBaseboard.EquipName,
-                                _,
+                                {},
```
Comment (Myoldmopar) [inline src/EnergyPlus/BaseboardElectric.cc:358]: So will this be the final outcome for this argument?  Or is this an intermediate step.  If we reordered the arguments would it make it easier to make them plain defaults?  Obviously these are just questions for a later time as we get this batch of effort merged in.

File: src/EnergyPlus/Furnaces.hh [hunk 3]
```diff
@@ -370,11 +370,11 @@ namespace Furnaces {
...
-                      Real64 &LatentLoadMet,     // Latent cooling load met (furnace outlet with respect to control zone humidity ratio)
-                      Real64 &OnOffAirFlowRatio, // Ratio of compressor ON mass flow rate to AVERAGE
-                      bool const HXUnitOn,       // flag to enable HX based on zone moisture load
-                      ObjexxFCL::Optional<Real64 const> CoolingHeatingPLRRat = _ // cooling PLR to heating PLR ratio, used for cycling fan RH control
+                      Real64 &SensibleLoadMet,                           // Sensible cooling load met (furnace outlet with respect to control zone temp)
+                      Real64 &LatentLoadMet,                             // Latent cooling load met (furnace outlet with respect to control zone humidity ratio)
+                      Real64 &OnOffAirFlowRatio,                         // Ratio of compressor ON mass flow rate to AVERAGE
+                      bool const HXUnitOn,                               // flag to enable HX based on zone moisture load
+                      Real64 const CoolingHeatingPLRRatio = 1.0          // cooling PLR to heating PLR ratio, used for cycling fan RH control
```
Comment (Myoldmopar) [inline src/EnergyPlus/Furnaces.hh]: :+1:

File: src/EnergyPlus/InternalHeatGains.cc [hunk 4]
```diff
@@ -8764,23 +8764,14 @@ namespace InternalHeatGains {
     }
 
     void UpdateInternalGainValues(EnergyPlusData &state,
-                                  ObjexxFCL::Optional_bool_const SuppressRadiationUpdate,
-                                  ObjexxFCL::Optional_bool_const SumLatentGains)
+                                  bool const SuppressRadiationUpdate,
+                                  bool const SumLatentGains)
```
Comment (Myoldmopar) [inline src/EnergyPlus/InternalHeatGains.cc]: :+1:

File: src/EnergyPlus/OutputProcessor.hh [hunk 5]
```diff
@@ -844,23 +844,23 @@ namespace OutputProcessor {
...
+                         std::string_view const EndUseKey = {},            // Meter End Use Key (Lights, Heating, Cooling, etc)
+                         std::string_view const EndUseSubKey = {},         // Meter End Use Sub Key (General Lights, Task Lights, etc)
+                         std::string_view const GroupKey = {},             // Meter Super Group Key (Building, System, Plant)
+                         std::string_view const ZoneKey = {},              // Meter Zone Key (zone name)
+                         int const ZoneMult = 1,                           // Zone Multiplier, defaults to 1
+                         int const ZoneListMult = 1,                       // Zone List Multiplier, defaults to 1
+                         int const indexGroupKey = -999,                   // Group identifier for SQL output
+                         std::string_view const customUnitName = {},       // the custom name for the units from EMS definition of units
+                         std::string_view const SpaceType = {}             // Space type (applicable for Building group only)
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputProcessor.hh:863]: :+1: :+1:

File: src/EnergyPlus/OutputProcessor.hh [hunk 6]
```diff
@@ -870,8 +870,8 @@ void SetupOutputVariable(EnergyPlusData &state,
                          OutputProcessor::SOVTimeStepType TimeStepTypeKey, // Zone, HeatBalance=1, HVAC, System, Plant=2
                          OutputProcessor::SOVStoreType VariableTypeKey,    // State, Average=1, NonState, Sum=2
                          std::string_view const KeyedValue,                // Associated Key for this variable
-                         ObjexxFCL::Optional_string_const ReportFreq = _,  // Internal use -- causes reporting at this freqency
-                         ObjexxFCL::Optional_int_const indexGroupKey = _   // Group identifier for SQL output
+                         std::string_view const ReportFreq = {},           // Internal use -- causes reporting at this freqency
+                         int const indexGroupKey = -999                    // Group identifier for SQL output
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputProcessor.hh:874]: OIC, the -999 isn't related to the default numeric for autosize.  I guess -999 is fine as a sentinel, but I'm sure there's a bit more cleaning that could be done on a future PR.

**Review by Myoldmopar — APPROVED** [review]
Good to go.  Made some comments, but nothing needs to be done here.

**Comment by Myoldmopar** [comment]
OK!  Jumping back into review to get the huge PR list back under control.  I think this one is an easy win.  I'll pull develop in and test to be sure.

**Comment by Myoldmopar** [comment]
All happy, tests pass, no format issues, and custom check is passing.  Merging, thanks @jasondegraw

**Pull request merged**